### PR TITLE
Add stdLibURL parameter to loadPyodide

### PR DESCRIPTION
**Pull request by rth**
Allows to customize the URL from which the Python standard library is loaded

Closes https://github.com/pyodide/pyodide/issues/3632

### CODE REVIEW

File: src/tests/test_pyodide.py [hunk 1]
```diff
@@ -1792,3 +1792,26 @@ def test_python_version(selenium):
...
+
+    try:
+        selenium.run_js(
+            """
+            let pyodide = await loadPyodide({
+                fullStdLib: false,
+                stdLibURL: "./python_stdlib2.zip",
+            });
+            // Check that we can import stdlib library modules
+            let math = pyodide.pyimport('math');
```
Comment (ryanking13) [inline src/tests/test_pyodide.py:1812]: `math` is a built-in library so I think we should test a different one.

File: src/js/module.ts [hunk 2]
```diff
@@ -162,7 +162,14 @@ function installStdlib(Module: Module, stdlibURL: string) {
  * @private
  */
 export function initializeFileSystem(Module: Module, config: ConfigType) {
-  installStdlib(Module, config.indexURL + "python_stdlib.zip");
+  let stdLibURL;
+  if (config.stdLibURL != undefined) {
+    stdLibURL = config.stdLibURL;
```
Comment (ryanking13) [inline src/js/module.ts:167]: Question: what happens if we pass relative urls?

**Review by ryanking13 — APPROVED** [review]
Thanks, @rth! Probably we need to document how `stdLibURL` and `fullStdLib` will work together later. Though for that, I guess we should first write a stable specification for repodata.json...

File: src/js/module.ts (hunk 2 above)
Comment (rth) [reply src/js/module.ts:167]: It just passes them to fetch, so relative URLs also work. That's what is tested currently.

File: src/tests/test_pyodide.py [hunk 3]
```diff
@@ -1792,3 +1792,26 @@ def test_python_version(selenium):
...
+    try:
+        selenium.run_js(
+            """
+            let pyodide = await loadPyodide({
+                fullStdLib: false,
+                stdLibURL: "./python_stdlib2.zip",
+            });
+            // Check that we can import stdlib library modules
+            let math = pyodide.pyimport('statistics');
+            assert(() => math.median([2, 3, 1]) === 2)
```
Comment (hoodmane) [inline src/tests/test_pyodide.py:1813]:
```suggestion
            let statistics = pyodide.pyimport('statistics');
            assert(() => statistics.median([2, 3, 1]) === 2)
```

**Review by WALID179 — APPROVED** [review]

**Pull request merged**